File: solutions/swift/prime-factors/1/Sources/PrimeFactors/PrimeFactors.swift
func primeFactors(_ number: Int64) -> [Int64] {
  // Write your code for the 'PrimeFactors' exercise in this file.
  if number <= 1 {
    return []
  } else {
    var p: Int64 = 2;
    var n = number
    var ans: [Int64] = []
    while n != 1 {
      while n % p != 0 {
        p += 1;
      }
      while n % p == 0 {
        ans.append(p)
        n /= p
      }
      p += 1
    }
    return ans
  }
}

File: solutions/csharp/prime-factors/2/PrimeFactors.cs
public static class PrimeFactors
{
    public static long[] Factors(long number)
    {
        var fs = new List<long>();
        long p = 2;
        var n = number;
        while (n != 1)
        {
            while (n % p != 0)
            {
                p += 1;
                if (p * p > n)
                {
                    p = n;
                }
            }
            while (n % p == 0)
            {
                n /= p;
                fs.Add(p);
            }
        }
        return [.. fs];
    }
}